src/commands/synthetics/__tests__/cli.test.ts

```diff
@@ -295,4 +300,109 @@ describe('run-test', () => {
       )
     })
   })
+
+  describe('exit code respects `failOnCriticalErrors`', () => {
+    test('404 test not found never exit with 1', async () => {
```

Comment:
Does this test case cover both non-critical errors "NO_TESTS_TO_RUN" and "NO_RESULTS_TO_POLL"? 🤔

Reply:
Added a test for `NO_RESULTS_TO_POLL` as well 👍